state/upgrades.go

```diff
@@ -180,3 +181,59 @@ func FillInEmptyCharmhubTracks(pool *StatePool) error {
 		return nil
 	}))
 }
+
+// AssignArchToContainers assigns an architecture to container
+// instance data based on their host.
+func AssignArchToContainers(pool *StatePool) error {
+	return errors.Trace(runForAllModelStates(pool, func(st *State) error {
+		instData, closer := st.db().GetCollection(instanceDataC)
+		defer closer()
+
+		var containers []instanceData
```

Comment:
In theory we'd not use instanceData struct directly, but it will be ok here since we're not going to change it.